portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,33 +37,54 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
-    public ArrayList<String> messages;
+  public ArrayList<String> messages = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // below command was used in servlet and fetch() tutorials
-    // response.setContentType("text/html;");
-    //
-    // below command was used in the servlet tutorial
-    // response.getWriter().println("<h1>Hello Gracie!</h1>");
-    //
-    // below command was used in the fetch() tutorial
-    // response.getWriter().println("Hello, Gracie is testing the fetch() function!");
-    //
     messages = new ArrayList<String>();
-    messages.add("Hello");
-    messages.add("Hi");
-    messages.add("Greetings");
-
-    String json = convertToJsonUsingGson(messages);
+    Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      messages.add(entity.getProperty("content").toString());
+    }
+    String json_comments = convertToJsonUsingGson(messages);
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(json_comments);
+  }
+
+  public String convertToJsonUsingGson(ArrayList<String> message_arraylist) {
+    Gson gson = new Gson();
+    String json = gson.toJson(message_arraylist);
+    return json;
   }
 
-  public String convertToJsonUsingGson(ArrayList<String> m) {
-      Gson gson = new Gson();
-      String json = gson.toJson(m);
-      return json;
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Retrieve inputs
+    String text = request.getParameter("text-input");
+    long time = System.currentTimeMillis();
+    
+    // Stores the result retrieved from the sentiment analyzer into a SentimentScore object
+    Document doc = Document.newBuilder().setContent(text).setType(Document.Type.PLAIN_TEXT).build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    float score = sentiment.getScore();
+    languageService.close();
+
+    // Print out the score value from the SentimentScore class
+    SentimentScore analyzed_score = new SentimentScore(score);
+    System.out.println(analyzed_score.CreateMessage());
+
+    // Setting properties of an entity object
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity textEntity = new Entity("Comment");
+    textEntity.setProperty("content", text);
+    textEntity.setProperty("time", time);
+    textEntity.setProperty("score", analyzed_score.GetScore());
```

Comment:
Ah, I see that you're already doing what I was suggesting! Great!

Reply:
I believe this is done in my next PR where I input the result in console.log().